File: API/Models/CartModel.cs
namespace ModelService
{
    public class CartModel
    {
        public int CustomerId { get; set; }
        public decimal? ItemSubTotal { get; set; }
        public decimal? TotalDiscountPrice { get; set; }
        public decimal? SubTotal { get; set; }
        public List<CartProductsModel>? Products { get; set; }
    }

    public class CartProductsModel
    {
        public int CartItemId { get; set; }
        public int ProductId { get; set; }
        public string? ProductImage { get; set; }
        public string? ProductName { get; set; }
        public string? Category { get; set; }
        public string? Brand { get; set; }
        public decimal? Discount { get; set; }
        public int StockQuantity { get; set; }
        public int Quantity { get; set; }
        public decimal? PriceBeforeDiscount { get; set; }
        public decimal? PriceAfterDiscount { get; set; }
    }

    public class CartUpdateModel
    {
        public int CartItemId { get; set; }
        public int Quantity { get; set; }
    }

    public class CartInsertModel
    {
        public int CustomerId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
    }
}

File: UI/src/app/shared/models/cart.model.ts
export interface CartModel {
    customerId: number;
    itemSubTotal?: number;
    totalDiscountPrice?: number;
    subTotal?: number;
    products?: CartProductsModel[];
}

export interface CartProductsModel {
    cartItemId: number;
    productId: number;
    productImage?: string;
    productName?: string;
    quantity: number;
    category?: string;
    brand?: string;
    discount?: number;
    stockQuantity?: number;
    priceBeforeDiscount?: number;
    priceAfterDiscount?: number;
}

export interface CartUpdateModel {
    cartItemId: number;
    quantity: number;
}

export interface CartInsertModel {
    customerId: number;
    productId: number;
    quantity: number;
}

